src/jep/convert_p2j.c

```diff
@@ -630,6 +735,8 @@ jobject PyObject_As_jobject(JNIEnv *env, PyObject *pyobject,
         return pyfastsequence_as_jobject(env, pyobject, expectedType);
     } else if (PyDict_Check(pyobject)) {
         return pydict_as_jobject(env, pyobject, expectedType);
+    } else if (PyCallable_Check(pyobject) && isFunctionalInterfaceType(env, expectedType)) {
```

Comment:
isFunctionalInterfaceType has alot of error conditions that return NULL,  we need error handling here also to ensure it propagates.